src/App.tsx

```diff
@@ -1,204 +1,101 @@
-/* eslint-disable jsx-a11y/control-has-associated-label */
-import React from 'react';
+import React, {
+  useCallback,
+  useEffect,
+  useMemo,
+  useState,
+} from 'react';
+import { getTodos } from './api/todos';
+import { Todo } from './types/Todo';
+import { Errors } from './types/Errors';
+import { Status } from './types/Status';
 import { UserWarning } from './UserWarning';
+import { TodoList } from './components/TodoList';
+import { Header } from './components/Header';
+import { Footer } from './components/Footer';
+import { ErrorNotification } from './components/ErrorNotification';
 
-const USER_ID = 0;
+const USER_ID = 96;
 
 export const App: React.FC = () => {
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [filterBy, setFilterBy] = useState(Status.all);
+  //  const [visibleTodos, setVisibleTodos] = useState(todos);
+  const [errorMessage, setErrorMessage] = useState<Errors | null>(null);
+
+  const onAllClick = useCallback(() => {
+    if (filterBy !== Status.all) {
+      setFilterBy(Status.all);
+    }
+  }, [filterBy]);
+
+  const onActiveClick = useCallback(() => {
+    if (filterBy !== Status.active) {
+      setFilterBy(Status.active);
+    }
+  }, [filterBy]);
+
+  const onCompletedClick = useCallback(() => {
+    if (filterBy !== Status.completed) {
+      setFilterBy(Status.completed);
+    }
+  }, [filterBy]);
+
+  const visibleTodos = useMemo(() => {
+    switch (filterBy) {
+      case Status.all:
+        return todos;
+
+      case Status.active:
+        return todos.filter(todo => !todo.completed);
+
+      case Status.completed:
+        return todos;
```

Comment:
I guess you don't finish `case Status.completed`

it should be
` return todos.filter(todo => todo.completed);`